### PR TITLE
Tones down the power of nitrous oxide, the reagent. Makes heparin a bit harder to fix to compensate

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Nitrous Oxide, rather than directly subtracting blood volume per tick, instead applies the anticoagulant trait ''TRAIT_BLOODY_MESS''. It shares this with heparin. 

However, unlike, heparin, coagulants like Sanguirite will remove the trait and allow for continued clotting while the reagent is present, neutering the nitrous oxide's anticoagulant effects (but not the other parts)

Heparin, on the other hand, will purge Sanguirite while it is in you system. You must remove the heparin before you can apply an anticoagulant.

## Why It's Good For The Game

Nitrous Oxide, on top of being a knockout chem that causes you to suffocate and become drowsy, just starts deleting blood rapidly. About 15 units of it, standard in a syringe, will kill you in about a minute, but you'll be unconscious for most of it (you'll be at around 50%-60% blood by the time it is out of your system, so as good as dead). It doesn't matter that it metabolizes quickly either, since because it isn't a toxin, it doesn't get purged by livers/improved livers, so you will probably metabolize all of the chem that enters your system.

Blood is one of those 'hidden damage types', a bit like brain damage. Once you start losing it _directly,_ you probably don't have a lot of options to resolve it (unlike a bleed, which you do in various manners), and the means of causing blood loss has been mostly pretty well controlled as of late. Heparin directly interacts with wounds as a good example.

Blood loss is also tied into oxyloss, which is another very mean damage type in that it causes you to fall into unconsciousness at 50 damage, so significantly more lethal than normal damage, kept in check by the fact that breathing restores some of it. Nitrous oxide, you might note, causes you to stop breathing.

It's cheaper to make than either heparin or lexorin, and since it isn't a toxin like those chems, it is able to circumvent a few game mechanics to simply just start killing you. It does the work of chloral hydrate, lexorin and heparin while it has a remarkably easy recipe.

Following the example of how lexorin was pulled into line, and consistency with heparin, I've made nitrous oxide an anticoagulant that may or may not come into play as one. I think this is more up to date with the state of toxins and chem warefare as a whole, and works with the relative ease in making nitrous oxide. And it has been this way for about 5 years, before we got wounds.

(did I mention that nitrous oxide is also an explosive if it gets hot enough?)

## TL;DR

I think a chem with a pretty basic recipe shouldn't be doing the work of 5 other, more complicated, chemicals while also not being a toxin and also not interacting with the wounds system or purity system whatsoever. And being an explosive.

## Changelog
:cl:
balance: Nitrous oxide, the reagent, increases bleed rate from wounds rather than directly subtracting blood. It can be counteracted using coagulants (such as those in epipens).
balance: Heparin purges coagulants. You have to remove heparin from someone's system before you can use coagulants.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -1480,11 +1481,24 @@
 		var/drowsiness_to_apply = max(round(reac_volume, 1) * 2 SECONDS, 4 SECONDS)
 		exposed_mob.adjust_drowsiness(drowsiness_to_apply)
 
+/datum/reagent/nitrous_oxide/on_mob_metabolize(mob/living/affected_mob)
+	if(!HAS_TRAIT(affected_mob, TRAIT_COAGULATING)) //IF the mob does not have a coagulant in them, we add the blood mess trait to make the bleed quicker
+		ADD_TRAIT(affected_mob, TRAIT_BLOODY_MESS, /datum/reagent/nitrous_oxide)
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1486]:
```suggestion
		ADD_TRAIT(affected_mob, TRAIT_BLOODY_MESS, type)
```

same result with less typing, and a bit more polite in case some deranged person decides to subtype this (they shouldn't though that's a bit weird). tres bon

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -1480,11 +1481,24 @@
...
+	return ..()
+
+/datum/reagent/nitrous_oxide/on_mob_end_metabolize(mob/living/affected_mob)
+	REMOVE_TRAIT(affected_mob, TRAIT_BLOODY_MESS, /datum/reagent/nitrous_oxide)
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1490]:
```suggestion
	REMOVE_TRAIT(affected_mob, TRAIT_BLOODY_MESS, type)
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 3]
```diff
@@ -1480,11 +1481,24 @@
...
+	return ..()
+
 /datum/reagent/nitrous_oxide/on_mob_life(mob/living/carbon/affected_mob, seconds_per_tick, times_fired)
 	affected_mob.adjust_drowsiness(4 SECONDS * REM * seconds_per_tick)
-	if(ishuman(affected_mob))
-		var/mob/living/carbon/human/affected_human = affected_mob
-		affected_human.blood_volume = max(affected_human.blood_volume - (10 * REM * seconds_per_tick), 0)
+
+	if(!HAS_TRAIT(affected_mob, TRAIT_BLOODY_MESS) && !HAS_TRAIT(affected_mob, TRAIT_COAGULATING)) //So long as they do not have a coagulant, if they did not have the bloody mess trait, they do now
+		ADD_TRAIT(affected_mob, TRAIT_BLOODY_MESS, /datum/reagent/nitrous_oxide)
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1497]:
```suggestion
		ADD_TRAIT(affected_mob, TRAIT_BLOODY_MESS, type)
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 4]
```diff
@@ -1480,11 +1481,24 @@
...
+
+	else if(HAS_TRAIT(affected_mob, TRAIT_COAGULATING)) //if we find they now have a coagulant, we remove the trait
+		REMOVE_TRAIT(affected_mob, TRAIT_BLOODY_MESS, /datum/reagent/nitrous_oxide)
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1500]:
```suggestion
		REMOVE_TRAIT(affected_mob, TRAIT_BLOODY_MESS, type)
```

**Review by san7890 — CHANGES_REQUESTED** [review]
The REALLY stupid thing is that nitrous oxide isn't an anticoagulant (it's a analgesic (painkiller)), they definitely meant _nitric_ oxide (which is a vasodilator, not an anticoagulant). whatever it's 2563 who cares (don't change it now)

**Comment by Spookuni** [comment]
Good change, N2O / ClF3 or N2O / Phlogiston syringes are currently an extreme double whammy death-mix with very little that comes close to their effective lethality.
The most dangerous toxin (which isn't actually a toxin, as Anne mentioned) in the game shouldn't also be an effective explosive that can delete targets from full health if they don't immediately extinguish themselves multiple times from chemical firestacking, to say nothing of the additional disruption drowsy + confusion + lose_breath stacking brings to the table.

This change drops N2O's combat potential down from "hilariously lethal and effective in all contexts" to "an effective disruptive chemical which can be hilariously lethal in specific circumstances" (Circumstances that other chemicals are notably better at) which I think is a much better place to stand

**Comment by necromanceranne** [comment]
Well I didn't touch the explosive part so you still gotta contend with that. I could do that but I kind of want to focus on the part I think is maybe the silliest and making the chem pretty braindead as a weapon.

**Comment by DustanHache** [comment]
> ## About The Pull Request
> Nitrous Oxide, rather than directly subtracting blood volume per tick, instead applies the anticoagulant trait ''TRAIT_BLOODY_MESS''. It shares this with heparin.
> 
> However, unlike, heparin, coagulants like Sanguirite will remove the trait and allow for continued clotting while the reagent is present, neutering the nitrous oxide's anticoagulant effects (but not the other parts)

as much as I hate this, I agree as a chemist main that it is extremely overpowered. unless you carry multiver on you, a single syringe shot is guarenteed lethal. on the flipside, heparin is not used for a reason, and that reason is that it is hot garbage. you have to already have wounds on the target for it to work (and if they have wounds to begin with, why not just keep hitting them?) and it requires formaldehyde, which the maintainers have taken a strong stance against producing (by requiring silver in the recipe as a primary ingredient)
for this reason, I believe that an alternative nerf should be considered unless it becomes a dumpster chem like acetone oxide and hydrogen peroxide were when their splash effects were borked.

**Comment by necromanceranne** [comment]
You're welcome to suggest something.

**Comment by DustanHache** [comment]
> You're welcome to suggest something.

my suggestion would be to instead nerf how fast it drains blood directly. the coefficient on it could probably be adjusted downward quite a bit to reduce lethality while still keeping it dangerous on high doses

**Comment by necromanceranne** [comment]
I don't that is at all tenable. The problem is that blood deletion is too efficient a killer, that is the primary reason I made this PR. I do not agree with it on principle as something an easy and very versatile chem like nitrous oxide should do. 

Blood loss is only _one_ use of this chem. I've touched nothing else.

**Comment by DustanHache** [comment]
> I don't think that is at all tenable. The problem is that blood deletion is too efficient a killer, that is the primary reason I made this PR. I do not agree with it on principle as something an easy and very versatile chem like nitrous oxide should do.
> 
> Blood loss is only _one_ use of this chem. I've touched nothing else.

the only other use it has is as a precursor chem. i would rather see its potency moved to heparin or even made into another toxin rather than have it erased from existence. hell, heparin would be actually worthwhile if it had the effect even in part (especially with the slow metabolism.)
the only real issue i see is that it is overtuned as it currently stands, and only because of *how fast* it goes from dose to lethality, with absolutely ludicrous drain rate for a rather mid tier difficulty chem.

**Comment by nukularpower** [comment]
its so damn rare to actually see a poison kill anymore.  does this shit really need to be nerfed

**Comment by DustanHache** [comment]
> syringe guns being deptfab locked. 

you cant even print syringe guns aside from the rapid fire syringe gun, and that one requires illegal tech.

**Comment by san7890** [comment]
> its so damn rare to actually see a poison kill anymore, between most toxins being "balanced" into being useless in a practical sense and syringe guns being deptfab locked. does this shit really need to be nerfed more

corag still exists

**Comment by nukularpower** [comment]
> > syringe guns being deptfab locked.
> 
> you cant even print syringe guns aside from the rapid fire syringe gun, and that one requires illegal tech.

thats what I was talking about, RF syringe gun is medifab only, and then ONLy if you ALSO have illegals

I just dont see the point of nerfing poisons at this point given how extremely rarely used they are

**Comment by necromanceranne** [comment]
Please don't make such grand statements of the decay of chem poisoning over a line change in one chem in particular. If you want other poisons to matter more, you are welcome to try your luck at a PR of your own.

**Comment by Loafin34** [comment]
> > syringe guns being deptfab locked.
> 
> you cant even print syringe guns aside from the rapid fire syringe gun, and that one requires illegal tech.

you ever heard about blowguns from bamboo? definitely underused but still.

**Comment by DustanHache** [comment]
> > > syringe guns being deptfab locked.
> > 
> > 
> > you cant even print syringe guns aside from the rapid fire syringe gun, and that one requires illegal tech.
> 
> you ever heard about blowguns from bamboo? definitely underused but still.

you realize they have a windup as well as other drawbacks? by the time you fire, you will have telegraphed your action for a full five seconds. thats unusable in 98% of situations, and certainly not usable for ant situation where toxins would be a goalz
